pkg/disttask/framework/taskexecutor/manager.go

```diff
@@ -223,15 +219,13 @@ func (m *Manager) onRunnableTasks(tasks []*proto.Task) {
 		logutil.Logger(m.logCtx).Info("detect new subtask", zap.Int64("task-id", task.ID))
 
 		if !m.slotManager.canAlloc(task) {
-			failpoint.Inject("taskTick", func() {
-				<-onRunnableTasksTick
-			})
+			logutil.Logger(m.logCtx).Warn("subtask has been rejected", zap.Int64("task-id", task.ID))
 			continue
 		}
 		m.addHandlingTask(task.ID)
+		m.slotManager.alloc(task)
```

Comment:
if it's moved out, we have to free on error

Reply:
yes👍

Reply:
https://github.com/pingcap/tidb/pull/49136/commits/06422f4c8ddcc8cc33ac0f1e2a2e5a70e99452a2